services/debugger/destination/eventDeliveryStatusUploader.go

```diff
@@ -27,73 +28,129 @@ type DeliveryStatusT struct {
 	EventType     string          `json:"eventType"`
 }
 
-var (
-	uploadEnabledDestinationIDs map[string]bool
-	configSubscriberLock        sync.RWMutex
-)
-
-var uploader debugger.Uploader[*DeliveryStatusT]
+type Opt func(*Handle)
 
-var (
+type Handle struct {
 	configBackendURL                  string
+	log                               logger.Logger
+	started                           bool
 	disableEventDeliveryStatusUploads bool
-	eventsDeliveryCache               debugger.Cache[*DeliveryStatusT]
-)
+	eventsDeliveryCache               cache.Cache[*DeliveryStatusT]
+	uploader                          debugger.Uploader[*DeliveryStatusT]
+	uploadEnabledDestinationIDs       map[string]bool
+	uploadEnabledDestinationIDsMu     sync.RWMutex
+	ctx                               context.Context
+	cancel                            func()
+	initialized                       chan struct{}
+	done                              chan struct{}
+}
+
+var WithDisableEventUploads = func(disableEventDeliveryStatusUploads bool) func(h *Handle) {
+	return func(h *Handle) {
+		h.disableEventDeliveryStatusUploads = disableEventDeliveryStatusUploads
+	}
+}
+
+func NewHandle(opts ...Opt) *Handle {
+	h := &Handle{
+		configBackendURL: config.GetString("CONFIG_BACKEND_URL", "https://api.rudderstack.com"),
```

Comment:
Can't we pass the backend url as a parameter instead of relying on `config`?

Reply:
Same for the logger.

Reply:
With the way the code is structured right now I guess it would need a significant amount of refactoring to ensure that we can pass them as parameters and moving away from config usages across the files. With this PR already significantly large does it make sense that we have it as a followup task which can be undertaken at a later point in time when we make changes with regards to apphandlers/runner?

Reply:
Can you please create a task to pick it up at the first convenience?

Reply:
[Added it here](https://www.notion.so/rudderstacks/8aac9087df644365acdf64e28e290153?v=6e06b0a5ade24f0aa5ffe05dc2972e84&p=0d205985825a4eddbc6c616c291f8fc7&pm=s)